### PR TITLE
Attempt to improve GitHub Pull Request template

### PR DESCRIPTION
**Pull request by ClobberXD**
- Remove redundant headings
- Convert majority of the text into comments
- Convert majority of the remaining text into headings

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-- Goal of the PR
-- How does the PR work?
-- Does it resolve any reported issue?
-- If not a bug fix, why is this PR needed? What usecases does it solve?
+### How does the PR work?
+<!-- Elaborate on the changes, if applicable, or delete this heading -->
 
-## To do
+### Use-cases
+<!-- What use-cases does this PR serve? Delete this heading if this is a bugfix PR -->
```
Comment (sfan5) [inline .github/PULL_REQUEST_TEMPLATE.md]: `<!-- Which use cases does this PR serve? Delete this heading if the PR does not implement new features -->`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,19 +1,24 @@
-Add compact, short information about your PR for easier understanding:
+<!-- Add compact, short information about your PR for easier understanding, like the objective of the PR -->
 
-- Goal of the PR
-- How does the PR work?
-- Does it resolve any reported issue?
-- If not a bug fix, why is this PR needed? What usecases does it solve?
+### How does the PR work?
```
Comment (sfan5) [inline .github/PULL_REQUEST_TEMPLATE.md]: I dont feel like the heading here is really needed

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
-- [ ] List
-- [ ] Things
-- [ ] To do
+<!-- Add "closes #<issue number>" or "fixes #<issue number>" if this PR resolves any reported issue(s) -->
 
-## How to test
+This PR is a Work in Progress
 
-<!-- Example code or instructions -->
+This PR is Ready for Review
```
Comment (sfan5) [inline .github/PULL_REQUEST_TEMPLATE.md]: dot at the end of sentence (both)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,19 +1,24 @@
...
+<!-- ^ Delete any one -->
+
+### To do
+
+<!-- If this PR is WIP, list the things that should be finished before the PR is ready for review -->
+
+### How to test
+
+<!-- Add example code or instructions to test the PR -->
```
Comment (sfan5) [inline .github/PULL_REQUEST_TEMPLATE.md]: `<!-- Add example code or instructions to help reviewers and interested people test the PR -->`

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (ClobberXD) [reply .github/PULL_REQUEST_TEMPLATE.md]: This heading encourages PR authors to elaborate upon implementation details. The comment below this line indicates that this heading is optional:
```markdown
<!-- Elaborate on the changes, if applicable, or delete this heading -->
```

Are you sure this heading should be removed?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (ClobberXD) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think `What` would be better suited than `Which` here, as "which" is only used only when there is a finite set of choices. In this context, using "which" would imply that we're presenting a list of use-cases after this sentence, for the PR author to choose from. On the other hand, "what" is more general, and does not imply the existence of limited options to choose from.

Apart from that _trivial_ nit, I've incorporated all your suggestions, incl. those from the comments below.

**Comment by rubenwardy** [comment]
I avoided comments because noobs tend to end up filling them in between the brackets

**Comment by HybridDog** [comment]
> I avoided comments because noobs tend to end up filling them in between the brackets

Maybe you can add a comment which links to a short markdown documentation.

**Comment by ClobberXD** [comment]
There aren't many noobs here, IMO. Raw text will make it more difficult for "non-noobs" as they have to delete most of the text manually. We can't make almost everyone's lives difficult just to make it easy for a couple of noobs, can we?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (sfan5) [reply .github/PULL_REQUEST_TEMPLATE.md]: i'm not <i>sure</i>, it's just personal preference
that said, the author can still be encouraged to elaborate even if the actual heading is gone

Comment (ClobberXD) [reply .github/PULL_REQUEST_TEMPLATE.md]: Makes sense. I'll remove it.

Comment (ClobberXD) [reply .github/PULL_REQUEST_TEMPLATE.md]: Most PRs are non-trivial, and elaborating on their implementation, which would at least serve as documentation, should be encouraged by default, IMO. And ofc, authors can always delete the heading if it doesn't make sense for their PR. Again, that's just my opinion, feel free to let me know if this should be removed.

**Comment by SmallJoker** [comment]
The current template is already quite good. Removing one or multiple lines isn't much of a deal and PR authors will change the contents of their posts to a large degree anyway.

**Comment by ClobberXD** [comment]
Alright, I'll close this PR. Thanks for the reviews and comments.

**Comment by ClobberXD** [comment]
Reopening, as I'm really finding it unnecessary to be deleting lines which can easily be converted into comments. While I admit I do largely modify the PR template to my liking, the template would serve its purpose well if it's a little easier to work with. This is a quality of life improvement, and doesn't have a good reason to be rejected. If this is low priority, I wholly understand - I can wait. :)

> PR authors will change the contents of their posts to a large degree anyway.

I'd reason that this is because the PR template is messy and requires a lot of fiddling anyway.

While there aren't too many lines to remove, I still have to manually go through the entire template and remove lines, which can be easily avoided if they were comments instead.

See #9901 for example, the To-do section is completely redundant. Instead of making the author delete all the `[ ] ----` lines, it'd be simpler to add a comment encouraging them to list the TODO stuff themselves, and hope that they provide the "list" in a legible way.

> I avoided comments because noobs tend to end up filling them in between the brackets

I feel like that's compromising the convenience of thousands of normal folks for the sake of a couple of people who can't find their way around GH/markdown, IMHO (apologies if the wording comes across as harsh - that's not my intention).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,19 +1,24 @@
...
+<!-- Add "closes #<issue number>" or "fixes #<issue number>" if this PR resolves any reported issue(s) -->
 
-## How to test
+This PR is a Work in Progress.
 
-<!-- Example code or instructions -->
+This PR is Ready for Review.
+<!-- ^ Delete any one -->
+
+### To do
```
Comment (ClobberXD) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: I noticed that I increased the headings' level by 1. While I think H3 suits these headings just fine, do let me know if I should revert these back to level 2 headings.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,19 +1,24 @@
...
 
-## To do
+### Use-cases
+<!-- What use cases does this PR serve? Delete this heading if the PR does not implement new features -->
 
-This PR is a Work in Progress / Ready for Review.
-<!-- ^ delete one -->
+****
 
-- [ ] List
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: I'd still encourage the classic to-do list due to the good GH integration. So maybe leave one `- [ ] TODO` line there?

Comment (ClobberXD) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: Most open PRs remove the checklist altogether (I checked :P). But I see why you think it's a good idea to include it in the template. I'll mention it in a comment instead, given that it's mostly removed.

**Pull request closed without merging**